### ISSUE TITLE
Port io submodule of QtCore

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/qtbase-os2#21]
Part of #19. This includes porting `QProcess`, `QLockFile`, `QStorageInfo`, and perhaps some pipe read/write helpers. Note that File-system related classes have been ported while working on `qmake` (see #5). The most complex case is `QProcess` and pipe stuff, the rest is relatively easy.

**Comment by dmik** [comment]
Note that the `QProcess` code committed above isn't tested yet, we will do it later (using Qt Tests) once QtCore builds. Same applies to other things like `QThread` (see #20).

**Comment by dmik** [comment]
Regarding `QLockFile`. Unfortunately, we won't be able to use POSIX advisory locking `fcntl(F_SETLK)` that we implemented in LIBCx for other Unix software. The reason is that in Qt `QLockFile` assumes that locking is effective between threads of the same process, not only between processes. But `fcntl(F_SETLK)` only provides the former (and this is how it is implemented in LIBCx). For this reason, Qt uses the BSD `flock` call on Unix. However, in kLIBC `flock` is implemented via a kLIBC version of `fcntl(F_SETLK)` which simply maps to `DosSetFileLocks` which also has a process granularity.

On Windows, they use file sharing flags to implement locking. We have them on OS/2 too but I need to check if it applies to threads. I guess only to processes. In this case we will have to use a mutex with each `QLockFile` to provide thread granularity or such.

**Comment by dmik** [comment]
According to my tests file share mode flags work well between threads as well and even within a single thread. I.e. if you attempt to open a second write handle to the same file which you denied writes to, this second open attempt will fail with ERROR_ACCESS_DENIED even in the same thread. I also wrote a multi-threaded tests and didn't get a single case were two threads would succeed opening such a file. This makes me think that this operation is atomic (although the docs only guarantee process granularity). So I went this way above.
